src/main/java/com/github/checkstyle/regression/Main.java

```diff
@@ -34,6 +74,160 @@ private Main() {
      * @throws Exception execute failure
      */
     public static void main(String[] args) throws Exception {
-        // empty for now
+        final Options options = createOptions();
+        final CommandLineParser parser = new DefaultParser();
+        final HelpFormatter formatter = new HelpFormatter();
+        CommandLine cmd = null;
+
+        try {
+            cmd = parser.parse(options, args);
+        }
+        catch (ParseException ex) {
+            System.err.println(ex.getMessage());
+            formatter.printHelp(null, options);
+            System.exit(1);
+        }
+
+        final Arguments arguments = ImmutableArguments.builder()
+                .repoPath(cmd.getOptionValue(OPT_LOCAL_GIT_REPO))
+                .branch(cmd.getOptionValue(OPT_PATCH_BRANCH))
+                .testerPath(cmd.getOptionValue(OPT_CHECKSTYLE_TESTER))
+                .stopAfterGenerateConfig(cmd.hasOption(OPT_STOP_AFTER_GENERATE_CONFIG))
+                .build();
+
+        validateArguments(arguments);
+        runRegression(arguments);
+    }
+
+    /**
+     * Creates and initializes the {@link Options} instance.
+     * @return the initialized options
+     */
+    private static Options createOptions() {
+        final Options options = new Options();
+
+        final Option repo = Option.builder("r")
+                .longOpt(OPT_LOCAL_GIT_REPO)
+                .required()
+                .hasArg()
+                .desc("the path of the checkstyle repository")
+                .build();
+        repo.setRequired(true);
+        options.addOption(repo);
+
+        final Option branch = Option.builder("p")
+                .longOpt(OPT_PATCH_BRANCH)
+                .required()
+                .hasArg()
+                .desc("the name of the PR branch")
+                .build();
+        options.addOption(branch);
+
+        final Option tester = Option.builder("t")
+                .longOpt(OPT_CHECKSTYLE_TESTER)
+                .required(false)
+                .hasArg()
+                .desc("the path of the checkstyle-tester directory")
+                .build();
+        options.addOption(tester);
+
+        final Option stopAfterGenerateConfig = Option.builder()
+                .longOpt(OPT_STOP_AFTER_GENERATE_CONFIG)
+                .required(false)
+                .hasArg(false)
+                .desc("indicates that regression tool would stop after generating config")
+                .build();
+        options.addOption(stopAfterGenerateConfig);
+
+        return options;
+    }
+
+    /**
+     * Validates the parsed CLI arguments.
+     * @param args the parsed CLI arguments.
+     * @throws IllegalArgumentException the arguments are invalid
+     */
+    private static void validateArguments(Arguments args) {
+        if (!existAndIsDirectory(args.repoPath())) {
+            throw new IllegalArgumentException(
+                    "path of local git repo must exist and be a directory");
```

Comment:
if I don't give repo path, this statement will look odd as it doesn't also say it is a required field.